test/ApiClient.test.ts

```diff
@@ -263,19 +268,18 @@ describe('Subscribe', () => {
       .mockImplementation(
         async (
           req: messageApi.SubscribeRequest,
-          cb: NotifyStreamEntityArrival<messageApi.Envelope> | undefined
+          cb: NotifyStreamEntityArrival<messageApi.Envelope> | undefined,
+          initReq?: InitReq
         ): Promise<void> => {
           called++
           if (called == 1) {
             throw new Error('error')
           }
-          for (let i = 0; i < 2; i++) {
-            if (cb) {
-              cb(createEnvelope())
-            }
-          }
+          let nonErroringSubscribe = subscribeMockImplementation(2)
```

Comment:
I don't fully understand what this change does. So first we start with a Subscribe mock that errors after one call, but before it gets there it'll instead return a non-erroring subscribe mock?

Does it still end up throwing the error and triggering a resubscribe?

Reply:
Okay I think I understand better. On the first subscribe request, this mock will invoke `nonErroringSubscribe` which should call `cb` with 2 messages. The second time `subscribe` is invoked it'll throw an error.

Reply:
This should match the original behavior, looks good.